Mathlib/LinearAlgebra/Matrix/Charpoly/Basic.lean

```diff
@@ -85,6 +86,11 @@ lemma charmatrix_fromBlocks :
   simp only [charmatrix]
   ext (i|i) (j|j) : 2 <;> simp [diagonal]
 
+lemma BlockTriangular.charmatrix {α : Type*} [Preorder α] {M : Matrix n n R} {b : n → α}
+    (h : M.BlockTriangular b) :
+    M.charmatrix.BlockTriangular b :=
+  (blockTriangular_diagonal _).sub fun i j h' => by simpa using h h'
```

Comment:
Is the reverse direction also true?

Reply:
Added the iff and reverse directions. It suggested four more lemmas about block triangular matrices, so I've added them to that file.